### DISCUSSION TITLE
jshell command

### DISCUSSION THREAD

**Discussion by ypiel-talend** [verhas/jamal#27] [Q&A] [answered]
In documentatoin it is written to simply execute:
> jshell https://git.io/jamal
This load the file https://raw.githubusercontent.com/verhas/jamal/master/jamal-cmd/jamal.jshell which defien a jamal version 1.6.5. he last one is 1.12.1.

It seems, last versions doesn't expect samearguments that the 1.6.5. If I download the jshel file, upgrade the version to 1.12.1 and execute it I have this error:

> ~/dev/github/perso/pocjamal$ jshell jamal.jshell  for.out.jam  
> [INFO] URL=https://repo.maven.apache.org/maven2/com/javax0/jamal/jamal-engine/1.11.1/jamal-engine-1.11.1.jar
> [..............]
> 1.11.1.jar:/home/ypiel/.m2/repository/com/javax0/jamal/jamal-core/1.11.1/jamal-core-1.11.1.jar:/home/ypiel/.m2/repository/com/javax0/jamal/jamal-cmd/1.11.1/jamal-cmd-1.11.1.jar javax0.jamal.cmd.JamalMain'
> Exception in thread "main" java.lang.IllegalArgumentException: You must specify at least two arguments: input and output
>         at javax0.jamal.cmd.JamalMain.main(JamalMain.java:106)

--help option seems doesn't have been updated. I don't know how to set output.

**Comment by ypiel-talend** [comment]
To be close as possible as previous way to execute the script, I have copied the jschell script in a gist: https://gist.githubusercontent.com/ypiel-talend/a978a171c5485dee1d49cac4bf0c0c07/raw/6cafd514b3f797ccc15f75426d8667afeda99ee4/jamal.jshell
and execute:

> jshell https://gist.githubusercontent.com/ypiel-talend/a978a171c5485dee1d49cac4bf0c0c07/raw/6cafd514b3f797ccc15f75426d8667afeda99ee4/jamal.jshell
> [INFO] URL=https://repo.maven.apache.org/maven2/com/javax0/jamal/jamal-engine/1.12.1/jamal-engine-1.12.1.jar
> [INFO] /home/ypiel/.m2/repository/com/javax0/jamal/jamal-engine/1.12.1/jamal-engine-1.12.1.jar is already in the repo
> [INFO] URL=https://repo.maven.apache.org/maven2/com/javax0/jamal/jamal-api/1.12.1/jamal-api-1.12.1.jar
> [INFO] /home/ypiel/.m2/repository/com/javax0/jamal/jamal-api/1.12.1/jamal-api-1.12.1.jar is already in the repo
> [INFO] URL=https://repo.maven.apache.org/maven2/com/javax0/jamal/jamal-tools/1.12.1/jamal-tools-1.12.1.jar
> [INFO] /home/ypiel/.m2/repository/com/javax0/jamal/jamal-tools/1.12.1/jamal-tools-1.12.1.jar is already in the repo
> [INFO] URL=https://repo.maven.apache.org/maven2/com/javax0/jamal/jamal-core/1.12.1/jamal-core-1.12.1.jar
> [INFO] /home/ypiel/.m2/repository/com/javax0/jamal/jamal-core/1.12.1/jamal-core-1.12.1.jar is already in the repo
> [INFO] URL=https://repo.maven.apache.org/maven2/com/javax0/jamal/jamal-cmd/1.12.1/jamal-cmd-1.12.1.jar
> [INFO] /home/ypiel/.m2/repository/com/javax0/jamal/jamal-cmd/1.12.1/jamal-cmd-1.12.1.jar is already in the repo
> [INFO] EXECUTING 'java -cp /home/ypiel/.m2/repository/com/javax0/jamal/jamal-engine/1.12.1/jamal-engine-1.12.1.jar:/home/ypiel/.m2/repository/com/javax0/jamal/jamal-api/1.12.1/jamal-api-1.12.1.jar:/home/ypiel/.m2/repository/com/javax0/jamal/jamal-tools/1.12.1/jamal-tools-1.12.1.jar:/home/ypiel/.m2/repository/com/javax0/jamal/jamal-core/1.12.1/jamal-core-1.12.1.jar:/home/ypiel/.m2/repository/com/javax0/jamal/jamal-cmd/1.12.1/jamal-cmd-1.12.1.jar javax0.jamal.cmd.JamalMain'
> Exception in thread "main" java.lang.IllegalArgumentException: You must specify at least two arguments: input and output
>         at javax0.jamal.cmd.JamalMain.main(JamalMain.java:106)

It seems it expects two arguments, but before it was processing all .jam files.

**Comment by verhas** [comment, marked as answer]
I think the new version processess the arguments differently and I forgot to update the JShell binding. I will fix it. In the meantime I recommend to use the jbang binding.

**Reply by ypiel-talend** [reply]
Ok, thanks,
Here it works :+1: 

> jbang jamal@verhas for.out.jam for.out
